test/lsp-mode-test.el

```diff
@@ -102,5 +102,81 @@
+    (clrhash lsp-diagnostic-stats)
+    (lsp--on-diagnostics workspace
+                         (lsp-make-publish-diagnostics-params
+                          :uri (lsp--path-to-uri "/foo/bar/baz/txt.txt")
+                          :diagnostics (vector
+                                        (lsp-make-diagnostic :severity? lsp/diagnostic-severity-error)
+                                        (lsp-make-diagnostic :severity? lsp/diagnostic-severity-warning))))
+    (should (equal (lsp-diagnostics-stats-for "/foo")
+                   [0 1 1 0 0]))
+    (should (equal (lsp-diagnostics-stats-for "/foo/bar")
+                   [0 1 1 0 0]))
+    (lsp--on-diagnostics workspace
+                         (lsp-make-publish-diagnostics-params
+                          :uri (lsp--path-to-uri "/foo/bar/baz/txt.txt")
+                          :diagnostics (vector
+                                        (lsp-make-diagnostic :severity? lsp/diagnostic-severity-error))))
+    (should (equal (lsp-diagnostics-stats-for "/foo")
+                   [0 1 0 0 0]))
+    (lsp--on-diagnostics workspace
+                         (lsp-make-publish-diagnostics-params
+                          :uri (lsp--path-to-uri "/foo/bar/baz/txt.txt")
+                          :diagnostics []))
+    (should (equal (lsp-diagnostics-stats-for "/")
+                   [0 0 0 0 0]))))
+
+(progn
+  ;; ert-deftest lsp-diagnostics-stats-windows-test ()
+  (let ((workspace (make-lsp--workspace))
+        (system-type 'windows-nt)
+        (lsp--uri-file-prefix "file:///"))
+    (clrhash lsp-diagnostic-stats)
+
+    (lsp--on-diagnostics workspace
+                         (lsp-make-publish-diagnostics-params
+                          :uri (lsp--path-to-uri "C:/foo/bar/baz/txt.txt")
+                          :diagnostics (vector
+                                        (lsp-make-diagnostic :severity? lsp/diagnostic-severity-error)
+                                        (lsp-make-diagnostic :severity? lsp/diagnostic-severity-warning))))
+
+    ;; (should (equal (lsp-diagnostics-stats-for "/foo")
+    ;;                [0 1 1 0 0]))
+    ;; (should (equal (lsp-diagnostics-stats-for "/foo/bar")
+    ;;                [0 1 1 0 0]))
+    ;; (lsp--on-diagnostics workspace
+    ;;                      (lsp-make-publish-diagnostics-params
+    ;;                       :uri (lsp--path-to-uri "/foo/bar/baz/txt.txt")
+    ;;                       :diagnostics (vector
+    ;;                                     (lsp-make-diagnostic :severity? lsp/diagnostic-severity-error))))
+    ;; (should (equal (lsp-diagnostics-stats-for "/foo")
+    ;;                [0 1 0 0 0]))
+    ;; (lsp--on-diagnostics workspace
+    ;;                      (lsp-make-publish-diagnostics-params
+    ;;                       :uri (lsp--path-to-uri "/foo/bar/baz/txt.txt")
+    ;;                       :diagnostics []))
+    ;; (should (equal (lsp-diagnostics-stats-for "/")
+    ;;                [0 0 0 0 0]))
```

Comment:
Why it is commented?

Reply:
I forgot to uncomment it(I used the uncommented portion for the benchmarks).